### PR TITLE
[PNUDEV-8] Security

### PR DESCRIPTION
**Pull request by volodymyrvynnyk**
image
image
image
image

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-security -->
```
Comment (Igor-Polatajko) [inline pom.xml:30]: Remove this line and dependency version (in spring boot we don't have to specify versions of spring dependencies, framework will do it for us)

File: src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                .antMatchers("/api/**")
+                .permitAll();
+    }
+
+    @Autowired
+    @SneakyThrows
+    public void configureGlobal(AuthenticationManagerBuilder auth) {
+        auth.inMemoryAuthentication()
+                .withUser(USERNAME)
+                .password(String.format("{noop}%s", PASSWORD))
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java:49]: Just wonder, why do we need this `{noop}` at the beginning?

File: src/main/java/com/pnu/dev/radioserviceapi/controller/admin/UserController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.pnu.dev.radioserviceapi.controller.admin;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import static java.util.Objects.nonNull;
+
+@Controller
+public class UserController {
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/dev/radioserviceapi/controller/admin/UserController.java:10]: AuthContoller

File: src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Value("${user.password}")
+    private String PASSWORD;
+
+    @Override
+    protected void configure(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity
+                //Admin security
+                .authorizeRequests()
+                .antMatchers("/admin/**")
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java:27]: check role here, if you added it to inmemory user

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -11,3 +11,8 @@ youtube:
   api_key: ${YOUTUBE_API_KEY}
   channel_id: ${YOUTUBE_CHANNEL_ID}
   max_results_number: 5
+
+
+user:
```
Comment (Igor-Polatajko) [inline src/main/resources/application.yaml:16]: admin_user

File: src/main/resources/templates/users/login.ftl [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<html>
+<head>
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css">
```
Comment (Igor-Polatajko) [inline src/main/resources/templates/users/login.ftl:3]: this dependencies have to be exctracted from header to some separate partial
no need to do it right now
just leave a ToDo comment here

File: src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java (hunk 2 above)
Comment (volodymyrvynnyk) [reply src/main/java/com/pnu/dev/radioserviceapi/config/WebSecurityConfig.java:49]: To command `AuthenticationManagerBuilder ` not to encode password

**Pull request merged**